src/proto/single/mod.rs

```diff
@@ -236,7 +240,7 @@ impl Job {
     }
 
     /// This job's id.
-    pub fn id(&self) -> &str {
+    pub fn id(&self) -> &JobId {
```

Comment:
shall we keep getters so that the end-users need less refactoring? or we can clean them up?